internal/commands/setup.go

```diff
@@ -198,6 +205,28 @@ func setWorkDir(specified string) error {
 	}
 }
 
+func downloadEnv(url string) (error, string) {
+	var envFile string
+	resp, err := httpClient.Get(url)
+	if err != nil {
+		return err, ""
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return err, ""
+		}
+		seconds := strconv.Itoa(int(time.Now().Unix()))
+		envFile = fmt.Sprintf("env-%s.yaml", seconds)
```

Comment:
I think we are better off writing this to tmp folder. Also might be useful to name the file based on content/remoteURI rather than a timestamp. A cleanup might also be desired after command finishes.

Reply:
I don't believe _any_ files should be written to disk. This causes cleanup problems (and no cleanup is happening as far as I can tell). I think the correct way to implement this is to have a couple of utility functions

* `isRemoteFile(file string) bool` - returns true if file starts with `http:` or `https:`
* `readEnvFile(file string)` - reads the file from local path or remote URL without creating any garbage to be cleaned up

In `setup.go` 

```
  abs, err := filepath.Abs(envFile)
```
should only be called for local files and [this function](https://github.com/splunk/qbec/blob/master/internal/model/app.go#L75) should be modified to read from local file or remote location.

Reply:
this also has the benefit of printing the URL in error messages on loading errors rather than some temp file name

Reply:
Good suggestions.  Let me give @gotwarlost's idea a try and see how it all fits together.